muffnn/autoencoder/autoencoder.py

```diff
@@ -426,36 +425,100 @@ def __getstate__(self):
+            def_indices -= set(self._sigmoid_indices.tolist())
+            self._sigmoid_msk_values[:, self._sigmoid_indices] = 1
+        self._sigmoid_msk_values = self._sigmoid_msk_values.astype(bool)
+
+        # Softmax indices.
+        if self.softmax_indices is not None:
+            self._softmax_indices \
+                = np.atleast_2d(self.softmax_indices)
+
+            self._softmax_msks_values = []
+            for begin, size in zip(self._softmax_indices[:, 0],
+                                   self._softmax_indices[:, 1]):
+                # Remove softmax stuff from def_indices.
+                def_indices -= set(range(begin, begin + size))
+
+                # Keep track of the mask.
+                msk = np.zeros((self.batch_size, self.input_layer_size_))
+                msk[:, range(begin, begin + size)] = 1
+                self._softmax_msks_values.append(msk)
 
-        # Mark the model as not fitted (i.e., not fully initialized based on
-        # the data).
-        self._is_fitted = False
+            self._softmax_msks_values \
+                = np.array(self._softmax_msks_values)
+        else:
+            self._softmax_indices = None
+            self._softmax_msks_values \
+                = np.empty((1, self.batch_size, self.input_layer_size_))
+        self._softmax_msks_values \
+            = self._softmax_msks_values.astype(bool)
+
+        # Default indices.
+        self._default_indices = np.array(list(def_indices), dtype=int)
+        self._default_msk_values \
+            = np.zeros((self.batch_size, self.input_layer_size_))
+        if len(self._default_indices) > 0:
+            self._default_msk_values[:, self._default_indices] = 1
+        else:
+            self._default_indices = None
+        self._default_msk_values = self._default_msk_values.astype(bool)
 
-        # Call partial fit, which will initialize and then train the model.
-        return self.partial_fit(X)
+        # Error check the results.
+        # We rebuild the sets here to error check the code above too.
+        # Always play good defense.
+        sigmoid_set = set()
+        if self._sigmoid_indices is not None:
+            sigmoid_set = set(self._sigmoid_indices.tolist())
+
+        softmax_set = set()
+        if self._softmax_indices is not None:
+            for begin, size in zip(self._softmax_indices[:, 0],
+                                   self._softmax_indices[:, 1]):
+                softmax_set |= set(range(begin, begin + size))
+
+        default_set = set()
+        if self._default_indices is not None:
+            default_set = set(self._default_indices.tolist())
+
+        if not sigmoid_set.isdisjoint(softmax_set):
+            raise ValueError("Sigmoid indices and softmax indices cannot"
+                             " overlap!")
+
+        if not sigmoid_set.isdisjoint(default_set):
+            raise ValueError("Sigmoid indices and default indices cannot"
```

Comment:
Yeah, these extra checks are def covered by the unit tests. I will adjust the code.